### PR TITLE
fix fill property with dot notation

### PR DESCRIPTION
**Pull request by zofe**
fix for https://github.com/livewire/livewire/issues/1776

### CODE REVIEW

File: src/ComponentConcerns/InteractsWithProperties.php [hunk 1]
```diff
@@ -136,14 +136,22 @@ public function propertyIsPublicAndNotDefinedOnBaseClass($propertyName)
...
-            if (in_array($key, $publicProperties)) {
-                $this->{$key} = $value;
+            if ($this->containsDots($key)) {
+                $propertyName = $this->beforeFirstDot($key);
+
+                $keyName = $this->afterFirstDot($key);
+                $targetKey = $this->beforeFirstDot($keyName);
+                data_set($this->{$propertyName}, $targetKey, $value);
+            } else {
+                if (in_array($key, $publicProperties)) {
```
Comment (iksaku) [inline src/ComponentConcerns/InteractsWithProperties.php:152]: You could remove the nesting level by using an `else if`

**Comment by iksaku** [comment]
Could you please revert your changes for the #1866 PR? They should be handled separately.

**Comment by danharrin** [comment]
Hey, thanks for your work here! Would you mind adding a couple of tests for this if you have time please? I'd suggest two:
- Filling a model property.
- Filling an associative array property.

**Comment by calebporzio** [comment]
Thanks for the PR. Just echoing what @danharrin said - mind adding tests for this new case?

**Pull request closed without merging**